### PR TITLE
Add support for arbitrary pokemon sprites

### PR DESCRIPTION
**Pull request by parkershepherd**
Pokemon icons are difficult to see on high resolution monitors, mobile devices, and tvs (at least on high zoom levels). I added a higher resolution icon set for these cases.

## Description
I extended the sprite functionality to allow arbitrary sprites to be selected, added a higher-resolution icon set, added the ability to scale the icon size, and auto-scaled icons with the map zoom level.

## Motivation and Context
Pokemon icons can be difficult to see on high resolution screens or tvs, especially at high zoom levels

## How Has This Been Tested?
Tested against Chrome/OSX, Firefox/OSX, Safari/OSX, Chrome/Android

## Screenshots (if appropriate):
screenshot at jul 24 20-40-56

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by hgentry** [comment]
This is really cool, I've been wanting bigger icons. Will test.

**Comment by theblackmarmoset** [comment]
How do I make mine have the option to chose between large and small icons?  I'm a noob with all this programming stuff.

File: static/sass/layout/_nav.scss [hunk 1]
```diff
@@ -59,6 +59,11 @@
 				}
 			}
 		}
+  
+    .form-control > select {
+      float: right;
+      font-size: 18px;
+    }
```
Comment (Chlodochar) [inline static/sass/layout/_nav.scss]: As much as I have the tabs in this file, I do have to say that the indentation doesn't match the surrounding lines.

File: templates/map.html [hunk 2]
```diff
@@ -102,6 +102,10 @@
 					</label>
 				</div>
 			</div>
+      <div class="form-control switch-container">
+        <h3>Icons</h3>
+        <select name="pokemon-icons" id="pokemon-icons"></select>
+      </div>
```
Comment (Chlodochar) [inline templates/map.html]: Same here.

File: static/map.js [hunk 3]
```diff
@@ -678,6 +691,21 @@ $('#scanned-switch').change(function() {
     }
 });
 
+$('#pokemon-icons').change(function() {
+    localStorage["pokemonIcons"] = this.value;
+    redrawPokemon(map_pokemons);
```
Comment (Chlodochar) [inline static/map.js]: Got an error here. `map_pokemons` is not defined in the latest develop. Other than that though, this looks pretty awesome.

Comment (Chlodochar) [inline static/map.js]: Needs to be:
```
    redrawPokemon(map_data.pokemons);
    redrawPokemon(map_data.lure_pokemons);
```

**Comment by parkershepherd** [comment]
@Chlodochar nice catches, updated accordingly

File: static/map.js [hunk 4]
```diff
@@ -290,16 +300,19 @@ function scannedLabel(last_modified) {
...
+    var offset = new google.maps.Point(
+        (index % sprite.columns) * sprite.width,
+        Math.floor(index / sprite.columns) * sprite.height);
+    return new google.maps.MarkerImage(sprite.filename, size, offset);
 }
 
 function setupPokemonMarker(item) {
-    var icon = new google.maps.MarkerImage("static/icons-sprite.png", new google.maps.Size(30, 30), calculateSpritePoints(parseInt(item.pokemon_id)));
+    var pokemon_index = item.pokemon_id - 1;
+    var icon = getGoogleSprite(pokemon_index, localStorage.pokemonIcons || 'default');
```
Comment (Chlodochar) [inline static/map.js]: It would be nice to not duplicate `'default'` in multiple spots, but this will be fixed once #1694 is merged.

**Comment by Chlodochar** [comment]
:+1: This looks good for merging.

**Comment by theblackmarmoset** [comment]
Is there support for iOS devices?

**Comment by parkershepherd** [comment]
@theblackmarmoset you'd have to wait for this to get merged (if it gets merged). It should have the same support for iOS that the develop branch has

**Comment by theblackmarmoset** [comment]
Alright. So as of now it's not possible to change the icon size on ios?

**Comment by parkershepherd** [comment]
@theblackmarmoset you could pull from my branch directly, but I don't plan on supporting it long term or anything

**Comment by theblackmarmoset** [comment]
How do I do that?  I literally know nothing about programming.  I already changed all the files to what they needed to be changed to and it still doesn't work on iOS for some reason

**Comment by chuyskywalker** [comment]
It'd be really awesome to switch to svg markers so we wouldn't need two config options for this, just, like, a slider for size. SVG would be the right resolution always.

**Comment by parkershepherd** [comment]
@chuyskywalker I agree, but my sprite generator doesn't seem to like SVGs very much. I could drop the low-res icons to remove that config option, but thought some people might want it for slightly lower bandwidth

**Comment by chuyskywalker** [comment]
@parkershepherd -- what generator? In my commit I just skipped spriting entirely. I don't think I've even ever heard of spriting SVGs...perhaps time for some research.

**Comment by parkershepherd** [comment]
@chuyskywalker I noticed a PR was merged using sprites instead of an icon folder and figured I wouldn't go against that. I did, however, find https://github.com/jkphl/iconizr which seems to work, although the resulting sprite is 4.6MB (1.6MB gzipped)

**Comment by chuyskywalker** [comment]
Ouch, that's probably too heavy. Let's not do that until, maybe, we can find a less huge SVG source.

**Comment by parkershepherd** [comment]
@chuyskywalker yeah, but svgs are very nice on computers/tvs, or even mobile devices if we have caching working (ETags, expires headers). My main problem is that iconizr doesn't respect the order of the icons, or my params to force a consistent size :/
image

**Comment by chuyskywalker** [comment]
I dunno, looks perfect to me/s

**Comment by Chlodochar** [comment]
Just to note: the coloring of the PNGs seem more fluent though. I like that more compared to the SVG coloring. Just my 2 cents.

**Comment by parkershepherd** [comment]
@chuyskywalker @Chlodochar any news on if this can be merged?

**Comment by Chlodochar** [comment]
I already voted on merging this :+1:. But if SVG needs to be added I'm fine waiting for that. This isn't a high impact change.

**Comment by chuyskywalker** [comment]
Test it out just now and it's super great looking. :+1:

**Comment by AHAAAAAAA** [comment]
+1

**Comment by Whoknowsit** [comment]
https://github.com/AHAAAAAAA/PokemonGo-Map/issues/1153

**Pull request merged**